test/powershell/Modules/Microsoft.PowerShell.Utility/Format-Custom.Tests.ps1

```diff
@@ -298,3 +298,135 @@ class MyLeaf2
+                        <CustomItem>
+                            <Text>Entry selected by ScriptBlock</Text>
+                        </CustomItem>
+                    </CustomEntry>
+                    <CustomEntry>
+                        <EntrySelectedBy>
+                            <SelectionCondition>
+                                <TypeName>MyTestObject</TypeName>
+                                <PropertyName>SelectProperty</PropertyName>
+                            </SelectionCondition>
+                        </EntrySelectedBy>
+                        <CustomItem>
+                            <Text>Entry selected by property</Text>
+                        </CustomItem>
+                    </CustomEntry>
+                    <CustomEntry>
+                        <CustomItem>
+                            <Text>Default was picked</Text>
+                        </CustomItem>
+                    </CustomEntry>
+                </CustomEntries>
+            </CustomControl>
+        </Control>
+    </Controls>
+    <ViewDefinitions>
+        <View>
+            <Name>DefaultView</Name>
+            <ViewSelectedBy>
+                <TypeName>MyTestObject</TypeName>
+            </ViewSelectedBy>
+            <GroupBy>
+                <PropertyName>Name</PropertyName>
+                <CustomControlName>MyTestControl</CustomControlName>
+            </GroupBy>
+            <TableControl>
+                <TableHeaders>
+                    <TableColumnHeader />
+                </TableHeaders>
+                <TableRowEntries>
+                    <TableRowEntry>
+                        <TableColumnItems>
+                            <TableColumnItem>
+                                <PropertyName>Name</PropertyName>
+                            </TableColumnItem>
+                        </TableColumnItems>
+                    </TableRowEntry>
+                </TableRowEntries>
+            </TableControl>
+        </View>
+    </ViewDefinitions>
+</Configuration>
+'@
+
+        Set-Content -Path $formatFilePath -Value $xmlContent
+        $ps = [powershell]::Create()
+        $iss = [initialsessionstate]::CreateDefault2()
+        $iss.Formats.Add($formatFilePath)
+        $rs = [runspacefactory]::CreateRunspace($iss)
+        $rs.Open()
+        $ps.Runspace = $rs
+    }
+
+    AfterAll {
+        $rs.Close()
+        $ps.Dispose()
+    }
+
+    It 'Property expression binding should be able to access the current object' {
+        $script = {
+            [PSCustomObject]@{
+                PSTypeName = 'MyTestObject'
+                SelectProperty = $true
+                Name = 'testing'
+            }
+        }
+
+        $null = $ps.AddScript($script).AddCommand('Out-String')
+        $ps.Streams.Error.Clear()
+        $expectedOutput = @'
+Entry selected by property
+
+Name
+----
+testing
+'@ -replace '\r?\n', [Environment]::NewLine
+
+        $ps.Invoke()[0].Trim() | Should -BeExactly $expectedOutput
```

Comment:
Why index 0 and trim?

Reply:
Index 0 because I know the collection will only have one item.  It's mostly just habit from methods that could exist on the collection as well like `ToString`, I can remove it if preferred.

Trim because white space isn't important to the test, and it saves some white space when declaring the expected output.  That's just cosmetic and can also be changed if preferred.